### PR TITLE
Migrate to a completely API compatible version of Artsy+UIFonts

### PR DESCRIPTION
**Pull request by orta**
CocoaPods 1.0 doesn't allow you to "weak link" between pods. 

In order to pull this off having two versions of our pods, they need to be the same from the perspective of CP, so we can say it's a dependency.

I'll highlight the changes, most of this PR is the `pod install` etc as we have checked in pods

### CODE REVIEW

File: Example/Tests/Tests.m [hunk 1]
```diff
@@ -7,7 +7,7 @@
 //
 
 #import <UIKit/UIKit.h>
-#import <Artsy+OSSUIFonts/UIFont+OSSArtsyFonts.h>
+#import <Artsy+UIFonts/UIFont+ArtsyFonts.h>
```
Comment (orta) [inline Example/Tests/Tests.m:10]: this is new

File: Artsy+UIFonts.podspec [hunk 2]
```diff
@@ -1,12 +1,12 @@
 Pod::Spec.new do |s|
-  s.name             = "Artsy+OSSUIFonts"
-  s.version          = "1.1.0"
+  s.name             = "Artsy+UIFonts"
```
Comment (orta) [inline Artsy+UIFonts.podspec:2]: So now, it's a lib on a different repo, but the same CocoaPods attributes

Comment (ashfurrow) [inline Artsy+UIFonts.podspec:2]: Will this conflict with the same pod name in the artsy Specs repo?

Comment (orta) [inline Artsy+UIFonts.podspec:2]: The version specified in the Podfile  will always take precedence

Comment (orta) [inline Artsy+UIFonts.podspec:2]: So, as long as we do the `:git =>` bit - it's OK

**Pull request merged**